release/make_history.py

```diff
@@ -23,84 +23,62 @@ def main():
-    commits = [c for c in commits if c]
-    print(f'found {len(commits)} commits since {last_tag}')
-    add_author(commits)
-    return commits
-
-
-@dataclass
-class Commit:
-    short_sha: str
-    message: str
-    pr: int
-    author: str | None = None
-
-    @classmethod
-    def from_line(cls, line: str) -> Commit | None:
-        short_sha, message = line.split(' ', 1)
-        message, last_word = message.rsplit(' ', 1)
-        m = re.fullmatch(r'\(#(\d+)\)', last_word)
-        if m:
-            pr = int(m.group(1))
-            return cls(short_sha, message, pr)
-
-    def __str__(self):
-        return f'{self.message} by @{self.author} in [#{self.pr}](https://github.com/pydantic/pydantic/pull/{self.pr})'
-
-
-def add_author(commits: list[Commit]) -> None:
-    print('Getting PR authors from GitHub...')
+    auth_token = get_gh_auth_token()
+
     session = requests.Session()
     headers = {
         'Accept': 'application/vnd.github+json',
+        'Authorization': f'Bearer {auth_token}',
         'x-github-api-version': '2022-11-28',
     }
-    missing = {c.pr for c in commits}
-    for page in range(1, 10):
-        print(f'getting GH pulls page {page}, looking for {len(missing)} missing authors...')
-        params = {'per_page': 100, 'page': page, 'direction': 'desc', 'sort': 'updated', 'state': 'closed'}
-        r = session.get('https://api.github.com/repos/pydantic/pydantic/pulls', headers=headers, params=params)
-        r.raise_for_status()
-        for pr in r.json():
-            pr_number = pr['number']
-            # debug(pr_number, missing, pr_number in missing)
-            if pr_number in missing:
-                for c in commits:
-                    if c.pr == pr_number:
-                        missing.remove(pr_number)
-                        c.author = pr['user']['login']
-                        break
-        if not missing:
-            print(f'all authors found after page {page}')
-            return
-        else:
-            print(f'{len(missing)} authors still missing after page {page}')
+
+    data = {'target_committish': 'dh/release-notes', 'previous_tag_name': last_tag, 'tag_name': f'v{new_version}'}
+    response = session.post(
+        'https://api.github.com/repos/pydantic/pydantic/releases/generate-notes', headers=headers, data=json.dumps(data)
+    )
+    response.raise_for_status()
+
+    body = response.json()['body']
+    body = body.removeprefix('<!-- Release notes generated using configuration in .github/release.yml at main -->\n\n')
+    body = body.removeprefix("## What's Changed\n")
+
+    body = re.sub(
+        pattern='https://github.com/pydantic/pydantic/pull/(\\d+)',
+        repl=r'[#\1](https://github.com/pydantic/pydantic/pull/\1)',
+        string=body,
+    )
+
+    # remove everything from "## New Contributors" and after - maybe add this
+    # back later
+    end = body.index('## New Contributors')
+    body = body[:end].strip()
```

Comment:
Here's what got skipped:

```
## New Contributors
* @15498th made their first contribution in [#7238](https://github.com/pydantic/pydantic/pull/7238)
* @GabrielCappelli made their first contribution in [#7213](https://github.com/pydantic/pydantic/pull/7213)
* @tobni made their first contribution in [#7184](https://github.com/pydantic/pydantic/pull/7184)
* @redruin1 made their first contribution in [#7282](https://github.com/pydantic/pydantic/pull/7282)
* @FacerAin made their first contribution in [#7288](https://github.com/pydantic/pydantic/pull/7288)
* @acdha made their first contribution in [#7297](https://github.com/pydantic/pydantic/pull/7297)
* @andree0 made their first contribution in [#7319](https://github.com/pydantic/pydantic/pull/7319)
* @gordonhart made their first contribution in [#7375](https://github.com/pydantic/pydantic/pull/7375)
* @pmmmwh made their first contribution in [#7496](https://github.com/pydantic/pydantic/pull/7496)
* @disrupted made their first contribution in [#7299](https://github.com/pydantic/pydantic/pull/7299)
* @prodigysml made their first contribution in [#7360](https://github.com/pydantic/pydantic/pull/7360)

**Full Changelog**: https://github.com/pydantic/pydantic/compare/v2.3.0...v2.4.0
```

We can keep it if you like it? It's not on previous history sections is all.

Reply:
@samuelcolvin